### ISSUE TITLE
add monthly coverage to monthly_means function

### ISSUE THREAD

**Issue by stephenholleran** [brightwind-dev/brightwind#1]
Add the option to return the monthly coverage for the monthly means in the monthly_means function.

option; return_coverage=False

monthly_plot, monthly_means = bw.monthly_means(df['Spd_80m'], return_data=True)
pd.concat([monthly_means, bw.coverage(df['Spd_80m'])], axis=1)

**Comment by Inder95** [comment]
There is a separate function coverage that returns coverage for any time-period specified. If we do this we will have two ways of doing the same thing which might be confusing for the user.

**Comment by stephenholleran** [comment]
Yes, I'm aware of the coverage function. As an analyst I will want to show the monthly mean wind speeds but I will also want beside the table of data the monthly coverage. I will then be able to tell why there is a difference in wind speeds due to coverage. To bring the means and coverage together I have to concat the tables together as above. We should make this easier by having an option to return this directly. I don't think this will confuse people.

**Comment by stephenholleran** [comment]
Doc string needs updating and missing  parameter and data type in doc string.
Link the commit so it can easily be seen what you did.

**Comment by stephenholleran** [comment]
Am getting a warning with this:
`df = bw.load_campbell_scientific(bw.datasets.demo_site_data)
bw.monthly_means(df['Spd80mS'], return_coverage=True)`

`C:\.........\brightwind\brightwind\transform\transform.py:64: UserWarning: Frequency of input data might not be determined correctly (most frequent time difference between adjacent timestamps does not match minimum time difference) most frequent time difference: 31 days 00:00:00  minimum time difference 28 days 00:00:00. Using most frequent time difference as resolution
  .format(pd.to_timedelta(most_freq_time_diff, unit='s'), minimum_time_diff))`

**Comment by Inder95** [comment]
The doc string was updated, the new parameter return_coverage is there with the data type. This warning comes because are calculating the frequency of data by comparing the minimum time difference and the most common time difference, when they are not equal this warning is shown.
The commit was linked above, I used a commit message Fixed #<issue no.> which automatically closes the issue and adds a link to that commit (might be handy going forward):
![image](https://user-images.githubusercontent.com/23189856/51107646-42604380-17e7-11e9-82f8-359498cf53c2.png)

**Comment by Inder95** [comment]
Can we add it as an enhancement in a separate issue? It would be good for tracking.

**Comment by stephenholleran** [comment]
You are too quick to close things. Let me do that :)

**Comment by stephenholleran** [comment]
Well. The coverage is grater than 1 so there is a bug here. For the dataset above.

**Comment by stephenholleran** [comment]
> Can we add it as an enhancement in a separate issue? It would be good for tracking.

OK, I'll create a new issue. Fix these bugs first though.

**Comment by stephenholleran** [comment]
> This warning comes because are calculating the frequency of data by comparing the minimum time difference and the most common time difference, when they are not equal this warning is shown.

I think this might disappear when you fix the greater than 1 coverage issue.

A better warning message would be better than what comes up.

**Comment by stephenholleran** [comment]
> The commit was linked above, I used a commit message Fixed # which automatically closes the issue and adds a link to that commit (might be handy going forward):

Ok great.

**Comment by Inder95** [comment]
This happens because of the algorithm we are using for calculating the resolution of time series. We take the minimum time difference between adjacent timestamps as mentioned in the warning. This creates problems with resolutions such as a month where the difference is not constant 30, 31 or 29 days. I need to look into a better algorithm or create a special create for such times periods. we discussed this problem earlier and it is good it is coming up again. I am changing the name of this issue to better track this.  I am tracking this in a new issue .

**Comment by stephenholleran** [comment]
Why have you closed this issue? From what I see there is a major bug?

**Comment by stephenholleran** [comment]
Are you sure the issue is with get_resolution and average_by_period functions? I have used the monthly_means and coverage functions independently and then merged them together and it works fine. Both of those use get_resolution and average_by_period I assume? Therefore I am expecting the issue is with the way you have combined the monthly mean and coverage in this monthly_mean function? It seems to be finding the time resolution on the monthly data which it shouldn't be doing. It should be finding the time resolution on the 10-min data? That is, you send it 10-min data, it should go get the monthly mean wind speed. The same 10-min data should also be sent to the coverage function. The two dataframes returned should then be merged together. As far as i see it that is how it should work unless there are some filtering that I have forgotten about.

**Comment by Inder95** [comment]
@stephenholleran  You are right this is not the issue here. But I will keep #16 open because it is an issue. For instance, when we send monthly time series and do yearly coverage that will create a problem.
